### PR TITLE
CO : Add the theme name to the compile_id

### PR DESCRIPTION
**Pull request by djbuch**
In order to avoid collisions between different themes (in multishop mode), we have to specify the theme name in the compile id.

| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | In order to avoid collisions between different themes (in multishop mode), we have to specify the theme name in the compile id. Otherwise we were loading compiled templates from other themes in the current theme randomly
| Type?         | bug fix
| Category?     | CO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | 
| How to test?  | Create a multi shop with different themes but sames modules activated. The compiled templates will be loaded from different themes if we do not set the compile ID.

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/20033)

### CODE REVIEW

**Review by PierreRambaud — APPROVED** [review]

**Review by Progi1984 — DISMISSED** [review]

**Comment by eternoendless** [comment]
Wouldn't this lead to collisions if two shops use the same theme? Maybe it should use the shop id instead?

File: config/smarty.config.inc.php
Comment (PierreRambaud) [inline config/smarty.config.inc.php]:
```suggestion
$smarty->setCompileId(Context::getContext()->shop->id . '-' . _THEME_NAME_);
```
@eternoendless is right, it doesn't concern the multishop, combined to the theme name is more efficient

**Review by PierreRambaud — CHANGES_REQUESTED** [review]
See comments

**Review by PierreRambaud — APPROVED** [review]

**Review by matthieu-rolland — APPROVED** [review]

**Review by Progi1984 — APPROVED** [review]

**Comment by ngodefroy** [comment]
currently blocked by https://github.com/PrestaShop/PrestaShop/issues/21066

**Comment by ngodefroy** [comment]
Hello @djbuch 

I still had mixed themes together in my case.

here is how I tested it:

In multistore, created different shops, added virtual url to each ones.
After that, I checked the different links which all redirected to different themes.
As you can see, One of my theme was mixed with the classic one:

On the right, is the original theme.
On the left, is the mixed theme I got when I switched from classic to the new theme virtual url.
Capture d’écran 2020-09-21 à 14 55 05

Thank you

**Comment by SimonGrn** [comment]
Hello,

there is no issue linked with this PR. Please create an issue.

**Comment by SD1982** [comment]
HI @djbuch I also still have mixed template with your PR

**I have this result:**
image

**It's a mix between this template**
image

**And this one:**
image

Also I don't see any issue linked with your PR, can you please add an issue to describe the bug ?

**Comment by SD1982** [comment]
Hi @djbuch as we got no answer in 30 days we close the PR [following our rules](https://devdocs.prestashop.com/1.7/project/maintainers-guide/reviewing-pull-requests/#stale-pull-requests) but if you come back we'll be happy to reopen it 
Have a nice day

**Comment by hadjedjvincent** [comment]
> Hi @djbuch as we got no answer in 30 days we close the PR [following our rules](https://devdocs.prestashop.com/1.7/project/maintainers-guide/reviewing-pull-requests/#stale-pull-requests) but if you come back we'll be happy to reopen it
> Have a nice day

Issue is available here : https://github.com/PrestaShop/PrestaShop/issues/20301

**Pull request closed without merging**
